Feat/fe web layout

Yehorik11 opened the pull request:
added layout heder

Yehorik11 committed:
style: 💄 header layout implementation

Implemented the layout for the header section with navigation and middle
section

BREAKING CHANGE: 🧨 Changed the structure of the `Header` component, which may require
updates to its usage in

✅ Closes:   other parts of the application. The `header` element is now wrapped
with a `div` and requires

Tornik73 commented:
Fix merge commits

Tornik73 commented on apps/fe-web/src/app/components/Account/Account.tsx:
```diff
@@ -0,0 +1,9 @@
+export default function Account() {
+  return (
+    <div className={styles.account}>
+      <p>Sign In</p>&nbsp;/&nbsp;<p>Sign Up</p>
```
We need to use localization

Tornik73 commented on apps/fe-web/src/app/components/CallNow/CallNow.module.scss:
```diff
@@ -0,0 +1,9 @@
+.callNow {
+  display: flex;
+  align-items: center;
+  .phoneNumber {
```
No need to use nested styles here

Tornik73 commented on apps/fe-web/src/app/components/Header/Header.module.scss:
```diff
@@ -0,0 +1,16 @@
+.wrapper {
+  width: 100%;
+  background-color: $gray-800;
+  .nav {
```
remove nested styles

Tornik73 commented on apps/fe-web/src/app/components/Header/Header.module.scss:
```diff
@@ -0,0 +1,16 @@
+.smallOne,
+.midle {
```
rename `midle`

Tornik73 commented on apps/fe-web/src/app/components/Header/Header.tsx:
```diff
@@ -0,0 +1,25 @@
+import CallNow from '../CallNow/CallNow';
```
add index.ts

Tornik73 commented on apps/fe-web/src/app/components/Midle/Midle.tsx:
```diff
@@ -0,0 +1,32 @@
+        <span className={styles.divider}>&nbsp;</span>
+        <Space size='small'>
+          <Badge count={4} showZero className={styles.customBadge}>
+            <Avatar shape='square' size='large' src={Bag.src} />
```
please create our custom for this logic, and I am not sure that is good idea to use avatar component for icons

Tornik73 commented on apps/fe-web/src/app/components/Midle/Midle.tsx:
```diff
@@ -0,0 +1,32 @@
+        </Space>
+        <div className={styles.info}>
+          <p>Shopping cart:</p>
+          <span>$57.00</span>
```
Create task in Jira for prices rendring, we should move this logic to the ui-shared lib

Tornik73 commented on apps/fe-web/src/app/components/NavLinks/NavLinks.tsx:
```diff
@@ -0,0 +1,27 @@
+
+const navLinks = [
+  { href: '/', label: 'Home', isActive: true },
+  { href: '/Shop', label: 'Shop' },
+  { href: '/Pages', label: 'Pages' },
+  { href: '/Blog', label: 'Blog' },
+  { href: '/About Us', label: 'About Us' },
+  { href: '/Contact Us', label: 'Contact Us' },
```
```suggestion
  { href: '/shop', label: 'Shop' },
  { href: '/pages', label: 'Pages' },
  { href: '/blog', label: 'Blog' },
  { href: '/about-us', label: 'About Us' },
  { href: '/contact-us', label: 'Contact Us' },
```

Tornik73 commented on apps/fe-web/src/app/components/SmallOne/SmallOne.tsx:
```diff
@@ -0,0 +1,31 @@
+            <Image src={DropDown} alt='Drop Down' />
+          </div>
+        </div>
+        <span className={styles.divider}>|</span>
```
create separate component for it

Tornik73 commented on apps/fe-web/src/app/components/SmallOne/SmallOne.tsx:
```diff
@@ -0,0 +1,31 @@
+
+import DropDown from '../../assets/images/Drop Down.svg';
+
+export default function SmallOne() {
```
I do not like this name

Tornik73 commented on apps/fe-web/src/app/components/SmallOne/SmallOne.tsx:
```diff
@@ -0,0 +1,31 @@
+
+      <div className={styles.links}>
+        <div className={styles.selector}>
+          <div className={styles.option}>
+            <p>Eng</p>
+            <Image src={DropDown} alt='Drop Down' />
+          </div>
```
should be in separate component

Tornik73 commented on apps/fe-web/src/app/components/SmallOne/SmallOne.tsx:
```diff
@@ -0,0 +1,31 @@
+            <Image src={DropDown} alt='Drop Down' />
+          </div>
+
+          <div className={styles.option}>
+            <p>USD</p>
+            <Image src={DropDown} alt='Drop Down' />
+          </div>
```
should be in separate component

Tornik73 commented on apps/fe-web/src/app/components/SmallOne/SmallOne.tsx:
```diff
@@ -0,0 +1,31 @@
+
+import styles from './SmallOne.module.scss';
+
+import DropDown from '../../assets/images/Drop Down.svg';
```
```suggestion
import DropDownIcon from '../../assets/images/dropdown-icon.svg';
```

Tornik73 commented on apps/fe-web/src/app/components/SmallOne/SmallOne.module.scss:
```diff
@@ -0,0 +1,31 @@
+.smallOne {
```
remove nested styles
Rename smallOne to another name

Yehorik11 committed:
Merge branch 'dev' into feat/fe-web-layout

Yehorik11 committed:
fix: 🐛 rename component and fix scss files

  rename component Midle to HeaderActions, removed nested styles file,
added new component InfoOptions

BREAKING CHANGE: 🧨   rename component Midle to HeaderActions, removed nested styles file,
added new component InfoOptions

Yehorik11 committed:
style: 💄 create Price component

created new component Price

BREAKING CHANGE: 🧨 created new component Price

✅ Closes: created new component Price

Tornik73 commented on apps/fe-web/src/app/components/CallNow/CallNow.tsx:
```diff
@@ -0,0 +1,18 @@
+import styles from './CallNow.module.scss';
+
+const phoneNumber = 2195550114;
+const phoneDisplay = '(219) 555-0114';
```
use https://www.npmjs.com/package/react-phone-number-input for number formatting

Tornik73 commented on apps/fe-web/src/app/components/CallNow/CallNow.tsx:
```diff
@@ -0,0 +1,18 @@
+  return (
+    <div className={styles.callNow}>
+      <Image src={Phone} alt='phone' />
+      <a href={`tel:${phoneNumber}`} className={styles.phoneNumber}>
```
Create separate shared PhoneNumber component

Tornik73 commented on apps/fe-web/src/app/components/Details/Details.tsx:
```diff
@@ -0,0 +1,23 @@
+      <Location />
+
+      <div className={styles.links}>
+        <div className={styles.selector}>
+          <InfoOption>Eng</InfoOption>
+          <InfoOption>USD</InfoOption>
+        </div>
```
should be separate component <CustomSelect/>

Tornik73 commented on apps/fe-web/src/app/components/Details/Details.tsx:
```diff
@@ -0,0 +1,23 @@
+          <InfoOption>Eng</InfoOption>
+          <InfoOption>USD</InfoOption>
+        </div>
+        <span className={styles.divider}>|</span>
```
should be separate component

Yehorik11 committed:
fix: 🐛 fix layout header

fix layout header

BREAKING CHANGE: 🧨 fix layout header

✅ Closes: fix layout header

Yehorik11 committed:
style: 💄 added component layout

added component layout

BREAKING CHANGE: 🧨 added component layout

✅ Closes: added component layout

Tornik73 commented on apps/fe-web/src/app/components/Button/Button.tsx:
```diff
@@ -0,0 +1,44 @@
+          styles[variant],
+          styles[fill],
+          styles[size],
+          isHovered &&
```
You can remove it, and do it with css only

Tornik73 commented on apps/fe-web/src/app/components/Cta/Cta.module.scss:
```diff
@@ -0,0 +1,19 @@
+  padding-top: 16px;
+
+  p {
+    padding: 0 !important;
```
Remove important usage

Tornik73 commented on apps/fe-web/src/app/components/NavLinks/NavLinks.tsx:
```diff
@@ -0,0 +1,28 @@
+const navLinks = [
+  { href: '/', label: 'Home', isOption: true, isActive: true },
+  { href: '/Shop', label: 'Shop', isOption: true },
+  { href: '/Pages', label: 'Pages', isOption: true },
```
Rename pages routes

Tornik73 commented on apps/fe-web/src/app/assets/images/Map Pin.svg:
```diff
@@ -0,0 +1,4 @@
+<svg width="17" height="20" viewBox="0 0 17 20" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Please do not use spaces in names, check all places
Rename all icons to lower case

Tornik73 commented on apps/fe-web/src/app/components/CallNow/CallNow.tsx:
```diff
@@ -0,0 +1,16 @@
+import Image from 'next/image';
+import Phone from '../../assets/images/PhoneCall.svg';
```
Assets importing should be in this format 
```ts
// index.ts
export { ReactComponent as PhoneIcon } from './phone-icon.svg';

// Component file
import { PhoneIcon } from '@assets/icons';

<PhoneIcon/>
```

Tornik73 commented on apps/fe-web/src/app/components/CallNow/CallNow.tsx:
```diff
@@ -0,0 +1,16 @@
+
+import styles from './CallNow.module.scss';
+
+const phoneNumber = 2195550114;
+const phoneDisplay = '(219) 555-0114';
```
```suggestion
const PHONE_NUMBER = 2195550114;
const PHONE_DISPLAY = '(219) 555-0114';
```

Tornik73 commented on apps/fe-web/src/app/components/CopyRight/CopyRight.tsx:
```diff
@@ -0,0 +1,9 @@
+export default function CopyRight() {
+  return (
+    <div className={styles.copyRight}>
+      <p>Ecobazar eCommerce &copy; 2021. All Rights Reserved</p>
```
use current year here

Tornik73 commented on apps/fe-web/src/app/components/Footer/Footer.module.scss:
```diff
@@ -0,0 +1,17 @@
+  }
+}
+
+.copy_payment {
```
Use one style of naming for css selectors

Tornik73 commented on apps/fe-web/src/app/components/InputSubscribe/InputSubscribe.module.scss:
```diff
@@ -0,0 +1,17 @@
+.inputField {
+  display: flex;
+
+  input {
+    position: relative;
+    margin-right: -80px;
+    width: 492px;
+    padding: 14px 24px;
+    border: 1px solid $gray-200;
+    border-radius: $border-radius-button;
+    outline: none;
+  }
+
+  .buttonInside {
+    z-index: 100;
+  }
+}
```
```suggestion
.wrapper {
  display: flex;
}


  .input {
    position: relative;
    margin-right: -80px;
    width: 492px;
    padding: 14px 24px;
    border: 1px solid $gray-200;
    border-radius: $border-radius-button;
    outline: none;
  }

  .button {
    z-index: 100;
  }
```

Tornik73 commented on apps/fe-web/src/app/components/InputSubscribe/InputSubscribe.tsx:
```diff
@@ -0,0 +1,13 @@
+export default function InputSubscribe() {
+  return (
+    <div className={styles.inputField}>
+      <input type='email' placeholder='Your email address' />
```
I checked figma, and I think we can create our `<CustomInput/>` component and use here

Tornik73 commented on apps/fe-web/src/app/components/Header/Header.module.scss:
```diff
@@ -0,0 +1,21 @@
+.nav {
+  display: flex;
+  justify-content: space-between;
+  max-width: 1440px;
```
flex: 1

Tornik73 commented on apps/fe-web/src/app/components/HeaderActions/HeaderActions.tsx:
```diff
@@ -0,0 +1,34 @@
+        <Image src={Heart} alt='Heart' />
+        <span className={styles.divider}>&nbsp;</span>
+        <Space size='small'>
+          <Badge count={4} showZero>
+            <Avatar shape='square' size='large' src={Bag.src} />
+          </Badge>
```
We should not use Avatar component for the Bag icon

Tornik73 commented on apps/fe-web/src/app/components/Logo/Logo.tsx:
```diff
@@ -0,0 +1,19 @@
+    <div className={styles.logo}>
+      <Image src={LogoIcon} alt='Ecobazar' />
+      <span className={colorText === 'white' ? styles.headingWhite : ''}>
+        <h6>Ecobazar</h6>
```
Move project name to the constant, or to the hook

Tornik73 commented on apps/fe-web/src/app/components/Logo/Logo.module.scss:
```diff
@@ -0,0 +1,14 @@
+.logo {
+  display: flex;
+  align-items: center;
+
+  h6 {
+    font-size: $heading-6;
+    padding-left: 8px;
+    margin: 0;
+  }
+
+  .headingWhite {
+    color: $white;
+  }
+}
```
```suggestion
.logo {
  display: flex;
  align-items: center;
}
.title {
    font-size: $heading-6;
    padding-left: 8px;
    margin: 0;
  }

  .white {
    color: $white;
  }
```

Tornik73 commented on apps/fe-web/src/app/components/Logo/Logo.tsx:
```diff
@@ -0,0 +1,19 @@
+  return (
+    <div className={styles.logo}>
+      <Image src={LogoIcon} alt='Ecobazar' />
+      <span className={colorText === 'white' ? styles.headingWhite : ''}>
```
```suggestion
      <span className={styles[colorText]}>
```

Tornik73 commented on apps/fe-web/src/app/components/NavItem/NavItem.tsx:
```diff
@@ -0,0 +1,41 @@
+  );
+
+  return (
+    <>
```
remove fragment

Tornik73 commented on apps/fe-web/src/app/components/NavItem/NavItem.tsx:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <>
+      <li className={`${styles.navItem} ${isActive ? styles.active : ''}`}>
+        <Link href={href}>
+          {!isOption ? (
+            label
+          ) : (
+            <div className={styles.labelWithDropdown}>
+              {label} {dropDown}
+            </div>
+          )}
+        </Link>
```
should be separate component

Tornik73 commented on apps/fe-web/src/app/components/NavItem/NavItem.module.scss:
```diff
@@ -0,0 +1,23 @@
+.navItem {
+  color: $gray-400;
+  font-weight: $weight-500;
+  font-size: $font-size-sm;
+  padding-right: 32px;
+}
+
+.labelWithDropdown {
+  display: flex;
+  align-items: center;
+  img {
+    margin-left: 8px;
+  }
+}
+
+.link {
+  display: flex;
+  align-items: center;
+}
+
+.active {
+  color: $white;
+}
```
```suggestion
.navItem {
  color: $gray-400;
  font-weight: $weight-500;
  font-size: $font-size-sm;
  padding-right: 32px;
}
.labelWithDropdown {
  display: flex;
  align-items: center;
  img {
    margin-left: 8px;
  }

.link {
  display: flex;
  align-items: center;
}

.active {
  color: $white;
}
```

Tornik73 commented on apps/fe-web/src/app/components/NavLinks/NavLinks.tsx:
```diff
@@ -0,0 +1,28 @@
+  return (
+    <ul className={styles.navList}>
+      {navLinks.map((link) => (
+        <NavItem
+          key={link.href}
+          href={link.href}
+          label={link.label}
+          isActive={link.isActive}
+          isOption={link.isOption}
+        />
```
```suggestion
        <NavItem
          {...link}
          key={link.href}
        />
```

Tornik73 commented on apps/fe-web/src/app/components/NavLinks/NavLinks.tsx:
```diff
@@ -0,0 +1,28 @@
+
+import styles from './NavLinks.module.scss';
+
+const navLinks = [
```
```suggestion
const NAV_LINKS = [
```

Tornik73 commented on apps/fe-web/src/app/components/Search/Search.module.scss:
```diff
@@ -0,0 +1,49 @@
+  }
+}
+
+.inputWrapper {
+  position: relative;
+  width: 400px;
+  height: 100%;
+}
+
+.inputWrapper::before {
+  content: '';
+  position: absolute;
+  left: 18px;
+  top: 50%;
+  transform: translateY(-50%);
+  background: url('../../assets/images/Search.svg') no-repeat center / contain;
+  width: 20px;
+  height: 20px;
+  pointer-events: none;
+}
+
```
```suggestion
.wrapper {
  position: relative;
  width: 400px;
  height: 100%;
  
  &:before {
    content: '';
  position: absolute;
  left: 18px;
  top: 50%;
  transform: translateY(-50%);
  background: url('../../assets/images/Search.svg') no-repeat center / contain;
  width: 20px;
  height: 20px;
  pointer-events: none;
  }
}

```

Tornik73 commented on apps/fe-web/src/app/components/Search/Search.tsx:
```diff
@@ -0,0 +1,21 @@
+        <input type='text' placeholder='Search' />
+      </div>
+      <button
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
```
Remove this logic and is hovered state, use styles approach

`&:hover {
    background-color: $tertiary;
  }`

Tornik73 commented on apps/fe-web/src/app/components/SocialMedia/SocialMedia.module.scss:
```diff
@@ -0,0 +1,17 @@
+  border-radius: 50%;
+}
+
+.activeIcon {
```
```suggestion
.active {
```

Tornik73 commented on apps/fe-web/src/app/components/SocialMedia/SocialMedia.module.scss:
```diff
@@ -0,0 +1,17 @@
+.socialIcons {
```
```suggestion
.icons {
```

Tornik73 commented on apps/fe-web/src/app/components/SocialMedia/SocialMedia.tsx:
```diff
@@ -0,0 +1,65 @@
+];
+
+export default function SocialMedia() {
+  const [hovered, setHovered] = useState<string | null>(null);
```
```suggestion
```

Tornik73 commented on apps/fe-web/src/app/components/SocialMedia/SocialMedia.tsx:
```diff
@@ -0,0 +1,65 @@
+      {socialIcons.map(({ name, url, icon: Icon }) => (
+        <div
+          key={name}
+          className={
+            hovered === name
+              ? `${styles.activeIcon} ${styles.icon}`
+              : styles.icon
+          }
+          onMouseEnter={() => setHovered(name)}
+          onMouseLeave={() => setHovered(null)}
```
```suggestion
```

Tornik73 commented on apps/fe-web/src/app/components/Subscribe/Subscribe.module.scss:
```diff
@@ -0,0 +1,13 @@
+.subscribeWrapper {
```
rename to wrapper
